diff --git a/quarot/kernels/gemm.cu b/quarot/kernels/gemm.cu
index b8f6643..621e235 100644
--- a/quarot/kernels/gemm.cu
+++ b/quarot/kernels/gemm.cu
@@ -1,5 +1,6 @@
 #include <cutlass/gemm/device/gemm.h>
 #include <gemm.h>
+#include <ATen/cuda/CUDAContext.h>
 
 
 
@@ -38,7 +39,7 @@ void matmul_host(
             {1,                                0}
     };
 
-    auto status = gemmOp(arguments);
+    auto status = gemmOp(arguments, nullptr, at::cuda::getCurrentCUDAStream());
 
     ensure(status == cutlass::Status::kSuccess,
            cutlassGetStatusString(status));
diff --git a/quarot/kernels/quant.cu b/quarot/kernels/quant.cu
index 84399d2..8b97494 100644
--- a/quarot/kernels/quant.cu
+++ b/quarot/kernels/quant.cu
@@ -1,4 +1,5 @@
 #include <quant.h>
+#include <ATen/cuda/CUDAContext.h>
 
 
 template<typename T>
@@ -90,7 +91,8 @@ void sym_dequant_host(const int32_t *q,
 {
     dim3 block{std::min<uint32_t>(cols, 16), std::min<uint32_t>(rows, 16)};
     dim3 grid{cdiv(cols, block.x), cdiv(rows, block.y)};
-    sym_dequantize_i32_f16_kernel<<<grid, block>>>(
+    cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+    sym_dequantize_i32_f16_kernel<<<grid, block, 0, stream>>>(
             q,
             scale_row, scale_col,
             rows, cols, x);
