diff --git a/kernels/setup.py b/kernels/setup.py
index 4e1a014..3d80f36 100644
--- a/kernels/setup.py
+++ b/kernels/setup.py
@@ -13,7 +13,7 @@ from torch.utils.cpp_extension import BuildExtension, CUDAExtension, CUDA_HOME
 ROOT_DIR = os.path.dirname(__file__)
 
 # Supported NVIDIA GPU architectures.
-SUPPORTED_ARCHS = {"8.0", "8.6", "8.7", "8.9", "9.0"}
+SUPPORTED_ARCHS = {"8.0", "8.9", "9.0"}
 
 # Compiler flags.
 CXX_FLAGS = ["-g", "-O3", "-fopenmp", "-lgomp", "-std=c++17", "-DENABLE_BF16"]
@@ -153,110 +153,110 @@ if nvcc_cuda_version >= Version("11.2"):
 
 ext_modules = []
 
-fused_attention_extension_fine_grained_dense = CUDAExtension(
-    name="omniserve_backend.fused_attention_fine_grained_dense",
-    sources=[
-        "csrc/fused_attention/fused_attention_fine_grained/dense_attention/fused_attention.cpp",
-        "csrc/fused_attention/fused_attention_fine_grained/dense_attention/decoderMaskedMultiheadAttention.cu",
-        "csrc/fused_attention/fused_attention_fine_grained/fine_grained_common/update_kv_cache.cu",
-        "csrc/fused_attention/common/input_metadata_helper.cu"
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_extension_fine_grained_dense)
-
-fused_attention_extension_fine_grained_sparse = CUDAExtension(
-    name="omniserve_backend.fused_attention_fine_grained_sparse",
-    sources=[
-        "csrc/fused_attention/fused_attention_fine_grained/sparse_attention/fused_attention.cpp",
-        "csrc/fused_attention/fused_attention_fine_grained/sparse_attention/decoderMaskedMultiheadAttention.cu",
-        "csrc/fused_attention/fused_attention_fine_grained/fine_grained_common/update_kv_cache.cu",
-        "csrc/fused_attention/common/input_metadata_helper.cu"
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_extension_fine_grained_sparse)
-
-
-fused_attention_extension_per_tensor_dense = CUDAExtension(
-    name="omniserve_backend.fused_attention_per_tensor_dense",
-    sources=[
-        "csrc/fused_attention/fused_attention_per_tensor/dense_attention/fused_attention.cpp",
-        "csrc/fused_attention/fused_attention_per_tensor/dense_attention/decoderMaskedMultiheadAttention.cu",
-        "csrc/fused_attention/fused_attention_per_tensor/per_tensor_common/update_kv_cache.cu",
-        "csrc/fused_attention/common/input_metadata_helper.cu"
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_extension_per_tensor_dense)
-
-
-fused_attention_extension_per_tensor_sparse = CUDAExtension(
-    name="omniserve_backend.fused_attention_per_tensor_sparse",
-    sources=[
-        "csrc/fused_attention/fused_attention_per_tensor/sparse_attention/fused_attention.cpp",
-        "csrc/fused_attention/fused_attention_per_tensor/sparse_attention/decoderMaskedMultiheadAttention.cu",
-        "csrc/fused_attention/fused_attention_per_tensor/per_tensor_common/update_kv_cache.cu",
-        "csrc/fused_attention/common/input_metadata_helper.cu"
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_extension_per_tensor_sparse)
-
-fused_attention_extension_pure_dense = CUDAExtension(
-    name="omniserve_backend.fused_attention_pure_dense",
-    sources=[
-        "csrc/fused_attention/fused_attention_pure_dense/fused_attention.cpp",
-        "csrc/fused_attention/fused_attention_pure_dense/decoderMaskedMultiheadAttention.cu",
-        "csrc/fused_attention/fused_attention_pure_dense/update_kv_cache.cu",
-        "csrc/fused_attention/fused_attention_pure_dense/input_metadata_helper.cu"
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_extension_pure_dense)
-
-
-fused_attention_ctx_pool_extension = CUDAExtension(
-    name="omniserve_backend.fused_attention_ctx_pool",
-    sources=[
-        "csrc/fused_attention/sparse_utils/ContextPool/pybind.cpp",
-        "csrc/fused_attention/sparse_utils/ContextPool/context_pool_kernel.cu",
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_ctx_pool_extension)
-
-
-fused_attention_selector_extension = CUDAExtension(
-    name="omniserve_backend.fused_attention_selector",
-    sources=[
-        "csrc/fused_attention/sparse_utils/KVPageSelector/fused_kv_page_selector.cpp",
-        "csrc/fused_attention/sparse_utils/KVPageSelector/KVPageSelector.cu",
-    ],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(fused_attention_selector_extension)
+# fused_attention_extension_fine_grained_dense = CUDAExtension(
+#     name="omniserve_backend.fused_attention_fine_grained_dense",
+#     sources=[
+#         "csrc/fused_attention/fused_attention_fine_grained/dense_attention/fused_attention.cpp",
+#         "csrc/fused_attention/fused_attention_fine_grained/dense_attention/decoderMaskedMultiheadAttention.cu",
+#         "csrc/fused_attention/fused_attention_fine_grained/fine_grained_common/update_kv_cache.cu",
+#         "csrc/fused_attention/common/input_metadata_helper.cu"
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_extension_fine_grained_dense)
+
+# fused_attention_extension_fine_grained_sparse = CUDAExtension(
+#     name="omniserve_backend.fused_attention_fine_grained_sparse",
+#     sources=[
+#         "csrc/fused_attention/fused_attention_fine_grained/sparse_attention/fused_attention.cpp",
+#         "csrc/fused_attention/fused_attention_fine_grained/sparse_attention/decoderMaskedMultiheadAttention.cu",
+#         "csrc/fused_attention/fused_attention_fine_grained/fine_grained_common/update_kv_cache.cu",
+#         "csrc/fused_attention/common/input_metadata_helper.cu"
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_extension_fine_grained_sparse)
+
+
+# fused_attention_extension_per_tensor_dense = CUDAExtension(
+#     name="omniserve_backend.fused_attention_per_tensor_dense",
+#     sources=[
+#         "csrc/fused_attention/fused_attention_per_tensor/dense_attention/fused_attention.cpp",
+#         "csrc/fused_attention/fused_attention_per_tensor/dense_attention/decoderMaskedMultiheadAttention.cu",
+#         "csrc/fused_attention/fused_attention_per_tensor/per_tensor_common/update_kv_cache.cu",
+#         "csrc/fused_attention/common/input_metadata_helper.cu"
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_extension_per_tensor_dense)
+
+
+# fused_attention_extension_per_tensor_sparse = CUDAExtension(
+#     name="omniserve_backend.fused_attention_per_tensor_sparse",
+#     sources=[
+#         "csrc/fused_attention/fused_attention_per_tensor/sparse_attention/fused_attention.cpp",
+#         "csrc/fused_attention/fused_attention_per_tensor/sparse_attention/decoderMaskedMultiheadAttention.cu",
+#         "csrc/fused_attention/fused_attention_per_tensor/per_tensor_common/update_kv_cache.cu",
+#         "csrc/fused_attention/common/input_metadata_helper.cu"
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_extension_per_tensor_sparse)
+
+# fused_attention_extension_pure_dense = CUDAExtension(
+#     name="omniserve_backend.fused_attention_pure_dense",
+#     sources=[
+#         "csrc/fused_attention/fused_attention_pure_dense/fused_attention.cpp",
+#         "csrc/fused_attention/fused_attention_pure_dense/decoderMaskedMultiheadAttention.cu",
+#         "csrc/fused_attention/fused_attention_pure_dense/update_kv_cache.cu",
+#         "csrc/fused_attention/fused_attention_pure_dense/input_metadata_helper.cu"
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_extension_pure_dense)
+
+
+# fused_attention_ctx_pool_extension = CUDAExtension(
+#     name="omniserve_backend.fused_attention_ctx_pool",
+#     sources=[
+#         "csrc/fused_attention/sparse_utils/ContextPool/pybind.cpp",
+#         "csrc/fused_attention/sparse_utils/ContextPool/context_pool_kernel.cu",
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_ctx_pool_extension)
+
+
+# fused_attention_selector_extension = CUDAExtension(
+#     name="omniserve_backend.fused_attention_selector",
+#     sources=[
+#         "csrc/fused_attention/sparse_utils/KVPageSelector/fused_kv_page_selector.cpp",
+#         "csrc/fused_attention/sparse_utils/KVPageSelector/KVPageSelector.cu",
+#     ],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(fused_attention_selector_extension)
 
 
 qgemm_w8a8_extension = CUDAExtension(
@@ -311,15 +311,15 @@ fused_extension = CUDAExtension(
 ext_modules.append(fused_extension)
 
 # Layer normalization kernels.
-layernorm_extension = CUDAExtension(
-    name="omniserve_backend.layernorm_ops",
-    sources=["csrc/layernorm.cpp", "csrc/layernorm_kernels.cu"],
-    extra_compile_args={
-        "cxx": CXX_FLAGS,
-        "nvcc": NVCC_FLAGS,
-    },
-)
-ext_modules.append(layernorm_extension)
+# layernorm_extension = CUDAExtension(
+#     name="omniserve_backend.layernorm_ops",
+#     sources=["csrc/layernorm.cpp", "csrc/layernorm_kernels.cu"],
+#     extra_compile_args={
+#         "cxx": CXX_FLAGS,
+#         "nvcc": NVCC_FLAGS,
+#     },
+# )
+# ext_modules.append(layernorm_extension)
 
 # Activation kernels.
 activation_extension = CUDAExtension(
