diff --git a/setup.py b/setup.py
index 320c05e..f6863e5 100644
--- a/setup.py
+++ b/setup.py
@@ -21,9 +21,8 @@ def remove_unwanted_pytorch_nvcc_flags():
 
 def get_cuda_arch_flags():
     return [
-        '-gencode', 'arch=compute_75,code=sm_75',  # Turing
         '-gencode', 'arch=compute_80,code=sm_80',  # Ampere
-        '-gencode', 'arch=compute_86,code=sm_86',  # Ampere
+        '-gencode', 'arch=compute_89,code=sm_89',  # Ampere
     ]
     
 def third_party_cmake():
@@ -44,7 +43,7 @@ def third_party_cmake():
     retcode = subprocess.call([pip, 'install', '-e', hadamard_dir])
 
 if __name__ == '__main__':
-    third_party_cmake()
+    # third_party_cmake()
     remove_unwanted_pytorch_nvcc_flags()
     setup(
         name='quarot',
@@ -59,8 +58,8 @@ if __name__ == '__main__':
                 ],
                 include_dirs=[
                     os.path.join(setup_dir, 'quarot/kernels/include'),
-                    os.path.join(setup_dir, 'third-party/cutlass/include'),
-                    os.path.join(setup_dir, 'third-party/cutlass/tools/util/include')
+                    os.path.join(setup_dir, '../cutlass/include'),
+                    os.path.join(setup_dir, '../cutlass/tools/util/include')
                 ],
                 extra_compile_args={
                     'cxx': [],
